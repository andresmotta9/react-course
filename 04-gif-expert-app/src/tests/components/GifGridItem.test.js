import React from 'react';
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Tests on GifGridItem component', () => {

    const title = 'A title';
    const url = 'https://localhost/image.jpg';

    let wrapper = shallow(<GifGridItem title={title} url={url} />);

    beforeEach( () => {
        wrapper = shallow(<GifGridItem title={title} url={url} />);
    });

    test('should show <GifGridItem /> correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have a paragraph with the title', () => {
        const parg = wrapper.find('p').text().trim();
        expect( parg ).toBe(title);
    });
    
    test('should the image url and the alt be the same as the component paramters', () => {
        const {src, alt} = wrapper.find('img').props();
        expect( src ).toBe(url);
        expect( alt ).toBe(title);
    });
    
    test('should have the class animate__fadeIn', () => {
        const containerClass = wrapper.find('div').hasClass('animate__fadeIn');
        expect( containerClass ).toBe( true );
    });
    
})  
