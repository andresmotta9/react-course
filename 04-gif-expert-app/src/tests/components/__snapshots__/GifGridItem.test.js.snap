// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifGridItem component should show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://localhost/image.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
